### PR TITLE
Implement datetime functions in velox/sparksql.

### PR DESCRIPTION
**Pull request by lviiii**

### CODE REVIEW

File: velox/substrait/SubstraitToVeloxExpr.h [hunk 1]
```diff
@@ -100,6 +100,21 @@ class SubstraitVeloxExprConverter {
...
+    {"SECOND", "second"},
+    {"MINUTE", "minute"},
+    {"HOUR", "hour"},
+    {"DAY", "day"},
+    {"DAY_OF_WEEK", "day_of_week"},
+    {"DAY_OF_YEAR", "day_of_year"},
+    {"MONTH", "month"},
+    {"QUARTER", "quarter"},
+    {"YEAR", "year"},
+    {"YEAR_OF_WEEK", "year_of_week"}
```
Comment (rui-mo) [inline velox/substrait/SubstraitToVeloxExpr.h:116]: Hi, are these names all for extract function? Maybe they should be added to Substrait at below place.

https://github.com/substrait-io/substrait/blob/main/extensions/functions_datetime.yaml#L10

File: velox/functions/sparksql/Register.cpp [hunk 2]
```diff
@@ -146,6 +146,74 @@ void registerFunctions(const std::string& prefix) {
...
+  registerFunction<YearFunction, int32_t, Timestamp>(
+      {prefix + "year"});
+  registerFunction<YearFunction, int32_t, TimestampWithTimezone>(
+      {prefix + "year"});
+  registerFunction<YearOfWeekFunction, int32_t, Date>(
+      {prefix + "year_of_week"});
+  registerFunction<YearOfWeekFunction, int32_t, Timestamp>(
+      {prefix + "year_of_week"});
+  registerFunction<YearOfWeekFunction, int32_t, TimestampWithTimezone>(
+      {prefix + "year_of_week"});
```
Comment (rui-mo) [inline velox/functions/sparksql/Register.cpp:216]: Are you going to open PR to Velox to add these functions to spark sql?

**Review by rui-mo — APPROVED** [review]
LGTM.

File: velox/functions/sparksql/Register.cpp (hunk 2 above)
Comment (lviiii) [reply velox/functions/sparksql/Register.cpp:216]: Yes, I've opened this PR(https://github.com/facebookincubator/velox/pull/3362)

File: velox/substrait/SubstraitToVeloxExpr.h (hunk 1 above)
Comment (lviiii) [reply velox/substrait/SubstraitToVeloxExpr.h:116]: You're right, I will open a PR to add these names.

**Pull request merged**